src/solver/lidproc.c

```diff
@@ -654,61 +659,64 @@ void biocellFluxRates(double x[], double f[])
         SurfaceInfil = MIN(SurfaceInfil, maxRate);
     }
 
-    //... storage & soil layers are full
-    else if ( soilTheta >= soilPorosity && storageDepth >= storageThickness )
+    else
     {
-        //... limiting rate is smaller of soil perc and storage outflow
-        maxRate = StorageExfil + StorageDrain;
-        if ( SoilPerc < maxRate )
+        //... storage & soil layers are full
+        if ( soilTheta >= soilPorosity && storageDepth >= storageThickness )
         {
-            maxRate = SoilPerc;
-            if ( maxRate > StorageExfil ) StorageDrain = maxRate - StorageExfil;
-            else
+            //... limiting rate is smaller of soil perc and storage outflow
+            maxRate = StorageExfil + StorageDrain;
+            if ( SoilPerc < maxRate )
             {
-                StorageExfil = maxRate;
-                StorageDrain = 0.0;
+                maxRate = SoilPerc;
+                if ( maxRate > StorageExfil ) StorageDrain = maxRate - StorageExfil;
+                else
+                {
+                    StorageExfil = maxRate;
+                    StorageDrain = 0.0;
+                }
             }
-        }
-        else SoilPerc = maxRate;
-
-        //... apply limiting rate to surface infil.
-        SurfaceInfil = MIN(SurfaceInfil, maxRate);
-    }
+            else SoilPerc = maxRate;
 
-    //... either layer not full
-    else if ( storageThickness > 0.0 )
-    {
-        //... limit storage exfiltration by available storage volume
-        maxRate = SoilPerc - StorageEvap + storageDepth*storageVoidFrac/Tstep;
-        StorageExfil = MIN(StorageExfil, maxRate);
-        StorageExfil = MAX(StorageExfil, 0.0);
+            //... apply limiting rate to surface infil.
+            SurfaceInfil = MIN(SurfaceInfil, maxRate);
+        }
 
-        //... limit underdrain flow by volume above drain offset
-        if ( StorageDrain > 0.0 )
+        //... either layer not full
+        else
         {
-            maxRate = -StorageExfil - StorageEvap;
-            if ( storageDepth >= storageThickness) maxRate += SoilPerc;
-            if ( theLidProc->drain.offset <= storageDepth )
+            //... limit storage exfiltration by available storage volume
+            maxRate = SoilPerc - StorageEvap + storageDepth*storageVoidFrac/Tstep;
```

Comment:
@LRossman, In previous applications, I have found that spurious oscillations occur when you use the full maxRate to limit the infiltration. A dampened maxRate has worked better. A dampening factor around 0.90-0.98 does the trick in many cases.

Reply:
Thanks @cbuahin for the insight. As this commit was made only to address documented bugs can we leave the introduction of damping for a future update.